### ISSUE TITLE
can not found generic type in struct

### ISSUE THREAD

**Issue by Liyixin95** [tqwewe/kameo#14]
```rust
#[derive(Actor)]
pub struct MyActor<T> {
    arena: Vec<T>,
}

#[messages]
impl<T: Send + 'static> MyActor<T> {
    #[message]
    pub fn add(&mut self, t: T) {}
}
```

and i get the compile error:
```
error[E0412]: cannot find type `T` in this scope
   |
24 |     pub fn add(&mut self, t: T) {}
   |                              ^ not found in this scope
```

**Comment by tqwewe** [comment]
Hi @Liyixin95 , I've fixed this in https://github.com/tqwewe/kameo/pull/17

It's quite complicated in code, since I need to go through the parameters to infer if the method is using the generic at all. But I got it working in the end

**Comment by Liyixin95** [comment]
> Hi @Liyixin95 , I've fixed this in #17
> 
> It's quite complicated in code, since I need to go through the parameters to infer if the method is using the generic at all. But I got it working in the end

Very appreciate for such a quick fix! It works for me!
